### ISSUE TITLE
Manual Push/Pull Force Push/Pull requires a refresh status first and no notifications for success/fail

### ISSUE THREAD

**Issue by codermjb** [Komodo/KomodoEdit#3224]
### Short Summary
When using Manual Push/Pull Force Push/Pull Komodo 10 on a fast Internet connection usually doesn't time out (it uploads just fine). On a slow Internet connection it freezes for the 30 second time out and then fails, or sometimes doesn't upload.

On Komodo 11, I have to Force push several times for it to push. My Internet speed is 190Mbps: http://beta.speedtest.net/result/6676763875

Is there a way we can get improvments made to this and/or notification (green success/red error) when we manually publish files? This way we don't have to wait around and keep checking to see if the file uploaded successfully.

### Steps to Reproduce
See above

### Expected results
Force Push and Force Pull to work more reliable and/or a notification showing the status.

### Actual results
Force push/pull sometimes fails and no notifications are shown.

### Platform Information

*Komodo Edit or IDE?* IDE
*Komodo Version?* 11.0.1 nightly
*Operating System (and version)?* Xubuntu 16.0.4.2 LTS

**Comment by cgchoffman** [comment]
Hey @mjbrancato, i'm interested in improvements to publishing.  I don't know what you're asking for though.  What aspect of Publishing are you talking about?  Pushing from Places?  Pushing from the Publishing dialog?  it's hard to answer without knowing what you're doing.  As far as I know it already does what you're asking for.

**Comment by mjbrancato** [comment]
@cgchoffman when you right click from the file in places and choose publishing, then force push. I didn't know that existed anywhere else, so the only place I know.

For me in Linux, there are no notifications for this. There weren't even notifications in Komodo 10 either.

**Comment by cgchoffman** [comment]
Thanks for clarifying.

**Comment by cgchoffman** [comment]
For now, if you want to confirm the state of your files open *Tools  > Publishing: Choose your config name* then see what it tells you.
To quickly open your current publishing config you can click the cloud icon in the Dynamic Toolbar on the left:
![publishing](https://user-images.githubusercontent.com/655154/31150556-e2faf066-a848-11e7-8b59-b2a09ade8e7d.GIF)


Thanks for filing this.  Looks like notifications aren't firing.  I'm pretty sure the code is in there somewhere but it looks like it's not working.

**Comment by mjbrancato** [comment]
I don't see where you see that. This is all I see: http://prntscr.com/gsxgyp

As for the status of the file, I see here (black is synced): http://prntscr.com/gsxh8p

And what screencast program is that? I am trying to get that for Linux.

**Comment by Defman21** [comment]
If Carey recorded that on Windows (I bet he did) - it's called LICEcap. There are Linux analogs though (SimpleScreenRecorder/OBS + mp4/webm to gif converter).

**Comment by Naatan** [comment]
SimpleScreenRecorder is awesome. UI is a bit messy, but it does the job well.

**Comment by cgchoffman** [comment]
That first screenshot you shared is *not* Places.  You said places in your comment.  That screenshot is the Publishing dialog.  So are we talking about Places?  Or the Publishing dialog?

Ahhh you're using a different colour scheme.  You'll have to look at what that color scheme has set for file statues: *Tools > color scheme editor > Interface (in dropdown on the left): Status: ...*  looks like status Foreground is the clean file status.

And that (awesome but terribly named )screencap tool is LICEcap.

**Comment by mjbrancato** [comment]
@cgchoffman I am using Places. I do not ever use this Publishing menu anymore because it doesn't work properly for me since it takes several hours to sync. So I just auto publish after save, and deploy via git when I am done.

I only showed the screenshot because you had me go there. I can't screenshot dialog boxes. Lightshot doesn't support that.

I will look into the colour schemes later.

And I think it is showing me notifications now, now that the localhost issue is sorted out.

**Comment by cgchoffman** [comment]
> I do not ever use this Publishing menu anymore because it doesn't work properly for me since it takes several hours to sync. 

Has it ever run through completely?  If it has never run through completely then publishing will not work.  You have to perform at least one sync.  That is likely why *push* isn't working for you.

**Comment by cgchoffman** [comment]
> I only showed the screenshot because you had me go there. I can't screenshot dialog boxes. Lightshot doesn't support that.

I misunderstood here.  Yes that is the screen I was suggesting you use.

**Comment by mjbrancato** [comment]
@cgchoffman I never ran it all the way, always had issues with it, and if your site is over 2GB then it will take years. I only use the autopublish file after save, and manual force push to force overwrite.

Pushing files does work for me (without needing to sync an entire site which I will never need to do, and even if so, GIT will deploy all of the files), that isn't the problem. It's just sometimes when there is a conflict, Force Push sometimes you have to do it a couple times. And then not having notifications, but I think I am getting notifications now after fixing localhost.

**Comment by Naatan** [comment]
@mjbrancato why not set up excludes for files you don't want synced?

**Comment by cgchoffman** [comment]
But how can I provide support on this now?  That's not how the tool was designed.  It has no idea about tfiles you haven't told it about (ie. synced them). 

And what @Naatan said, if you're syncing 2GB of files you're including way too much it sounds like.

**Comment by mjbrancato** [comment]
@Naatan @cgchoffman I have been using it this way for over a year. That is why you implemented the force push feature is to push a file manually.

Even then, some files fail in the publishing box.

@Naatan I want any file I am currently working on to be published (and it does as defined in the save to autopublish).

**Comment by cgchoffman** [comment]
> That is why you implemented the force push feature is to push a file manually.

That's not quite true.  It was implemented so you can avoid opening the dialog every time to resolve a conflict.  You still have to perform an initial sync.  There is no way around that.  It won't work properly otherwise.  I honestly can't support this workflow.  We would havae to rewrite Publishing to support not performing an actual sync.

**Comment by cgchoffman** [comment]
> @Naatan I want any file I am currently working on to be published (and it does as defined in the save to autopublish).

@Naatan is not saying "don't push the files you're working on".  He's saying when you configure the publishing setup use the *excludes* field to remove folders that might have images or framework core modules that you shoudln't even be editing (depending on what you're working on) or what ever else.

**Comment by mjbrancato** [comment]
@cgchoffman what does the initial sync do in Publishing tools? I never have done that, and that never ever worked for me.

Force push always works for me even without doing the 2 hour sync that fails all the time from FTP timeout issues. All I mentioned is that sometimes the FTP times out when you manually push a file and there is no message coming back stating so.

**Comment by mjbrancato** [comment]
I am good then, I will stick to the way I am doing it because it works. I don't want to add excludes, I want to push whatever I want to push to the server including updated images. I have GIT that syncs the dev server from develop branch, then master to live server.

All I was asking for was a notification of failed file transfer, which can happen for anything.

**Comment by cgchoffman** [comment]
> I don't want to add excludes

we're just trying to help @mjbrancato :)  No need to get defensive.  This would likely make your life a lot easier if you to listen to our suggestion.  Please consider this fact if you come across further issues in the publishing tool.  

> All I was asking for was a notification of failed file transfer, which can happen for anything.

This is still a good point and I reproduced this so we'll keep the ticket.  Thank you.

**Comment by mjbrancato** [comment]
@cgchoffman I am not getting defensive, sorry. I was just saying I don't need that feature. And it would be hours for each hundreds of saves/changes I do every few minutes.

All I need to use is this: http://prntscr.com/gsxuy0

File uploads, but sometimes there is a conflict, so I go in and force push... sometimes the force push has a FTP timeout (You still have to force push in the publishing box too). Even if you use the Publishing tool in the exact way you mentioned to upload a file for each modification you do, you can still have a FTP timeout, FTP is flawed that way and some servers have some messed up FTP. So a notification is really all that is mainly needed.

I think I confused you guys with the way I explained it. Sorry about that.

Thank you.

**Comment by cgchoffman** [comment]
Ok gotcha, thanks for clarifying.

**Comment by codermjb** [comment]
So the Push/Force Push issue can be resolved with a workaround by doing a refresh status then triggering a push/force push.

No Cloud: http://prntscr.com/gvt0lo
Refresh Status Cloud Shows up: http://prntscr.com/gvt0ud
Force Push/Push now uploads: http://prntscr.com/gvt10k

**Comment by cgchoffman** [comment]
This was a confusing ticket.  I saw something mentioned a few times that I'd like to clarify about using publishing.

> And it would be hours for each hundreds of saves/changes I do every few minutes.

The publishing dialog does not sync every file.  It syncs what changed.  That is why I encouraged waiting out the initial sync.  After that it's just syncing the files that changed since last sync.  

Any way, i'm gonna take a look at the publishing stuff and try to see why it's not tell users when a push failed.
